Week3/homework/step2-6.js

```diff
@@ -4,13 +4,14 @@ const arr2d = [[1, 2], [3, 4], [5, 6]];
 const arr3d = [[[1, 2], [3, 4]], [[5, 6], [7, 8]]];
 
 function flattenArray2d(arr) {
-  // Replace this comment and the next line with your code
-  console.log(arr);
+  return arr.reduce((flattenedArr, currentArr) => [...flattenedArr, ...currentArr], []);
 }
 
 function flattenArray3d(arr) {
-  // Replace this comment and the next line with your code
-  console.log(arr);
+  return arr.reduce(
+    (flattenedArr, current2dArr) => [...flattenedArr, ...flattenArray2d(current2dArr)],
+    [],
+  );
```

Comment:
👍 for logic and using `flattenArray2d` inside `flattenArray3d`. However, I would recommend that you also read about a topic in programming called `recursion`.

Reply:
I tried to make it recursive, I did it but there might be a better way because I didn't like my solution to be honest.